assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/status/InfoRequest.java

```diff
@@ -0,0 +1,55 @@
+package org.eclipse.sw360.antenna.frontend.compliancetool.sw360.status;
+
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360Connection;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+abstract public class InfoRequest<T> {
+    abstract public String getInfoParameter();
+
+    boolean hasAdditionalParameters() {
+        return getAdditionalParameters().size() > 0;
+    }
+
+    abstract String helpMessage();
+    abstract boolean isValid();
+    abstract Set<String> getAdditionalParameters();
```

Comment:
Idea: _getAdditionalParameters()_ could return more information than just parameter names, e.g. data objects with the parameter name, the help text, and a flag whether the parameter is optional/mandatory. Then a major part of the parsing could be already implemented by the base class, also the generation of the help message.